webhooks/appmesh/virtualnode_validator.go

```diff
@@ -95,6 +102,14 @@ func (v *virtualNodeValidator) checkVirtualNodeBackendsForDuplicates(vn *appmesh
 	return nil
 }
 
+func (v *virtualNodeValidator) checkForRequiredFields(vn *appmesh.VirtualNode) error {
```

Comment:
Looks good. Let's also update the API docs [1] to align with the expected behavior [2] that if there's no inbound traffic, service discovery is optional. Updating the docs will be reflected here [3] when we release so customers are aware of the behavior


[1] https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/apis/appmesh/v1beta2/virtualnode_types.go#L335
[2] https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_nodes.html
[3] https://aws.github.io/aws-app-mesh-controller-for-k8s/reference/api_spec/#appmesh.k8s.aws/v1beta2.VirtualNodeSpec